### ISSUE TITLE
Owncloud reports PHP version 5.3.29 on a server running 5.4.45 and stops

### ISSUE THREAD

**Issue by dgorrd** [owncloud/core#26958]
### Steps to reproduce
1. Run the setup php file OR
2. Finish installation and go to next step, then
3. Owncloud failes saying that PHP version 5.3.29 is running when Hostgator says 5.4.45 is running.

### Expected behaviour
Tell us what should happen
It should go to the next step of setup without failing

### Actual behaviour
Tell us what happens instead
The index.php file aborts reporting PHP version error

### Server configuration
**Operating system**: Linux

**Web server:**

**Database:**

**PHP version:**

**ownCloud version:** (see ownCloud admin page)
latest

**Updated from an older ownCloud or fresh install:**

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```there is no integrity/ folder inside the owncloud/settings folder

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder
```

**The content of config/config.php:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

or 

Insert your config.php content here
(Without the database password, passwordsalt and secret)
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**

**Operating system:**

### Logs
#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by kdslkdsaldsal** [comment]
@dgorrd You can put a file with the following content side-by-side to the ownCloud files:

```
<?php
phpinfo();
?>
```

and open it in your browser. This is very very likely showing the same outdated PHP version. The setup check of oC is using native PHP methods for getting the PHP version so normally this can't be wrong.

**Comment by DeepDiver1975** [comment]
This is for sure an environment issue - please seek out for help on https://central.owncloud.org - THX

**Comment by dgorrd** [comment]
I know this is an environment issue but I was hoping Owncloud would do a 
better job of detecting the environment and figuring out that PHP 5.4.45 
was installed, not 5.3.29.  It doesn't seem possible now to use Owncloud 
on this web host.

**Comment by dgorrd** [comment]
When I did this it reports PHP 5.4.45 which is good.  But the 
installation files report 5.3.29 and abort.


Thanks,

Doug

**Comment by kdslkdsaldsal** [comment]
@dgorrd Just some background info:

ownCloud is using a native PHP constant ``PHP_VERSION`` (https://secure.php.net/manual/en/reserved.constants.php#reserved.constants.core) to get the running PHP version. Whats returned by this function is the actual running PHP which is executing the setup / php file

If you get a higher version from the phpinfo.php:

1. you have put this file to another location where a higher PHP version is running the script
2. your hoster is messing around in some other way with the execution of PHP scripts

Overall its the best to contact the support of your hoster. They should know why their environment is behaving like this.

**Comment by dgorrd** [comment]
I put the installer file into the same location as the phpinfo.php file, 
but they still report two separate versions. I did contact the website 
hosting company but once the tech support person saw phpinfo showing 
5.4.45 they refused to look any further.  They use a PHP version system 
where you select (rather than install) the PHP version you want and it 
goes into a configuration file.  I don't know if that is typical for web 
hosting companies to do that now, and I'm not sure what to do next 
except maybe look for another web hosting company.

**Comment by kdslkdsaldsal** [comment]
@dgorrd You can point them to a file containing:

```
<?php
echo("PHP version reported by PHP_VERSION is: " . PHP_VERSION);
?>
```

Overall ownCloud is using this constant as written before:

https://secure.php.net/manual/en/reserved.constants.php#reserved.constants.core

so ownCloud just can see whats coming back from this. There is nothing what can be done from oC side if ownCloud is getting a wrong version from your setup.

**Comment by dgorrd** [comment]
I think the environment problems are even deeper than that.  I altered 
the setup files to eliminate the check for PHP_VERSION and the setup 
crashed a little farther on when accessing the base.php file.  I tried 
to tell hostgator that, but they took one look at phpinfo.php reporting 
5.4.45 and refused to look any further. They use a system where the 
current php version is altered in a configuration file rather than a new 
installation of php... and the files are all hidden by aliases.
